### PR TITLE
Improve `Capybara::Node::Matchers` API doc

### PR DESCRIPTION
**Pull request by ybiquitous**
- Fix plain-text or broken links (references).
- Mark code text as possible in Markdown style.
- Fix a bit mistakes of text.

[skip ci]

The latest doc is:
https://www.rubydoc.info/gems/capybara/3.21.0/Capybara/Node/Matchers

### CODE REVIEW

**Comment by ybiquitous** [comment]
[question] Is it on purpose that `#assert_not_matches_selector` has no doc?

https://github.com/teamcapybara/capybara/blob/4bc8101afc664d15ddafc8ea11ec262fd9258365/lib/capybara/node/matchers.rb#L565-L566

https://www.rubydoc.info/gems/capybara/3.21.0/Capybara/Node/Matchers#assert_not_matches_selector-instance_method

File: lib/capybara/node/matchers.rb [hunk 1]
```diff
@@ -160,16 +160,16 @@ def assert_all_of_selectors(*args, **options, &optional_filter_block)
 
       # Asserts that none of the provided selectors are present on the given page
       # or descendants of the current node. If options are provided, the assertion
-      # will check that each locator is present with those options as well (other than :wait).
+      # will check that each locator is present with those options as well (other than `:wait`).
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: "is present" => "is not present"

File: lib/capybara/node/matchers.rb [hunk 2]
```diff
@@ -250,11 +250,11 @@ def assert_no_selector(*args, &optional_filter_block)
       # This will check if the expression occurs exactly 4 times.
       #
       # It also accepts all options that {Capybara::Node::Finders#all} accepts,
-      # such as :text and :visible.
+      # such as `:text` and `:visible`.
       #
       #     page.has_xpath?('.//li', text: 'Horse', visible: true)
       #
-      # has_xpath? can also accept XPath expressions generate by the
+      # {#has_xpath?} can also accept XPath expressions generate by the
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: generate => generated

**Comment by twalpole** [comment]
> [question] Is it on purpose that `#assert_not_matches_selector` has no doc?

Not that I remember - I would guess it just got missed.

**Comment by ybiquitous** [comment]
Thanks, I'll try to add the missing doc. 💪 

And I pushed minor fixes via 363375c694b9901fe6b0485f16b266fea9530131.

**Comment by ybiquitous** [comment]
I pushed the addition via 5e188da3a88b9c6720fde16f76fc29651825da8e. Please review again. 🙏 

After approved, I'll rebase the commits in the PR.

File: lib/capybara/node/matchers.rb [hunk 3]
```diff
@@ -19,21 +19,20 @@ module Matchers
...
       #
       #     page.has_selector?(:xpath, XPath.descendant(:p))
       #
       # @param (see Capybara::Node::Finders#all)
-      # @param args
-      # @option args [Integer] :count (nil)     Number of times the text should occur
-      # @option args [Integer] :minimum (nil)   Minimum number of times the text should occur
-      # @option args [Integer] :maximum (nil)   Maximum number of times the text should occur
-      # @option args [Range]   :between (nil)   Range of times that should contain number of times text occurs
+      # @option options [Integer] :count (nil)     Number of times the text should occur
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: Number of matching elements that should exist

File: lib/capybara/node/matchers.rb [hunk 4]
```diff
@@ -19,21 +19,20 @@ module Matchers
...
+      # @option options [Integer] :minimum (nil)   Minimum number of times the text should occur
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: Minimum number of matching elements that should exist

File: lib/capybara/node/matchers.rb [hunk 5]
```diff
@@ -19,21 +19,20 @@ module Matchers
...
+      # @option options [Integer] :maximum (nil)   Maximum number of times the text should occur
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: Maximum number of matching elements that should exist

File: lib/capybara/node/matchers.rb [hunk 6]
```diff
@@ -19,21 +19,20 @@ module Matchers
...
+      # @option options [Range]   :between (nil)   Range of times that should contain number of times text occurs
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: Range of number of matching elements that should exist

**Comment by twalpole** [comment]
I'm going to make the necessary changes I noticed, and will then squash and merge this

**Comment by ybiquitous** [comment]
Oh, thanks!

**Pull request merged**